src/scripts/developer.ts

```diff
@@ -0,0 +1,86 @@
+import { validateDeveloper, generateId } from "./helpers/index.js";
+
+export function addDeveloper(developers: Developer[], developer: Partial<Developer>) {
+	const isDevValidated = validateDeveloper(developer);
+	if (isDevValidated) {
+		const validatedDev: Developer = {
+			...developer,
+			id: generateId(developers),
+		} as Developer;
+		developers.push(validatedDev);
+
+		alert(`Developer ${developer.name} added`);
+	}
+}
+
+export function cloneDeveloper(developers: Developer[], developer: Developer): Developer {
+	const deepCopy: Developer = {
+		...JSON.parse(JSON.stringify(developer)),
+		id: generateId(developers),
+	};
+	return deepCopy;
+}
+
+export function updateDeveloper(developer: Developer, updates: Partial<Developer>) {
```

Comment:
1. do not directly modify `developer`, in most cases function should not directly modify its arguments, clone the argument, update and return 
2. code can be refactored, this works but in modern JS we have better and cleaner options than this